recipes/flatcc/all/conanfile.py

```diff
@@ -0,0 +1,115 @@
+    license = "Apache-2.0"
+    options = { "shared": [True, False],
+                "fPIC": [True, False],
+                "portable": [True, False],
+                "gnu_posix_memalign": [True, False],
+                "runtime_lib_only": [True, False],
+                "verify_assert": [True, False],
+                "verify_trace": [True, False],
+                "reflection": [True, False],
+                "native_optim": [True, False],
+                "fast_double": [True, False],
+                "ignore_const_condition": [True, False],
+    }
+    default_options = { "shared": True,
+                        "fPIC": True,
+                        "portable": False,
+                        "gnu_posix_memalign": True,
+                        "runtime_lib_only": False,
+                        "verify_assert": False,
+                        "verify_trace": False,
+                        "reflection": True,
+                        "native_optim": False,
+                        "fast_double": False,
+                        "ignore_const_condition": False
+    }
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "cmake"
+    exports_sources = ["CMakeLists.txt"]
+
+    _cmake = None
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            if self.settings.compiler == "Visual Studio" and self.options.shared:
+                #Building flatcc shared libs with Visual Studio is broken
+                raise ConanInvalidConfiguration("Building flatcc libraries shared is not supported")
+            if self.settings.compiler == "gcc":
+                raise ConanInvalidConfiguration("Building flatcc with MinGW is not supported")
+        if self.settings.os == "Macos" and self.options.shared:
+            raise ConanInvalidConfiguration("Building flatcc shared on Macos is currently not supported due to Systems Integrity Protection problems")
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["FLATCC_PORTABLE"] = self.options.portable
+            self._cmake.definitions["FLATCC_GNU_POSIX_MEMALIGN"] = self.options.gnu_posix_memalign
+            self._cmake.definitions["FLATCC_RTONLY"] = self.options.runtime_lib_only
+            self._cmake.definitions["FLATCC_INSTALL"] = True
+            self._cmake.definitions["FLATCC_COVERAGE"] = False
+            self._cmake.definitions["FLATCC_DEBUG_VERIFY"] = self.options.verify_assert
```

Comment:
the options name is very_assert, but the definition is debug_verify, what's the correct? Follow the same definition name if possible. What does this definition do?

Reply:
That last question was exactly why I changed the option name. In my opinion, from the option name you should more or less know what it does without having to dig into the cmake files or source code. But no problem for me to change the option name to 'debug_verify'.

From the CMakeLists.txt:

> # Normally a verify should just quickly reject invalid buffers but for troubleshooting, assertions can enabled.

Reply:
makes sense, so don't need to change.